solr/core/src/java/org/apache/solr/security/AuthenticationPlugin.java

```diff
@@ -52,11 +80,73 @@
   public abstract boolean doAuthenticate(ServletRequest request, ServletResponse response,
       FilterChain filterChain) throws Exception;
 
+  /**
+   * This method is called by SolrDispatchFilter in order to initiate authentication.
+   * It does some standard metrics counting.
+   */
+  public final boolean authenticate(ServletRequest request, ServletResponse response, FilterChain filterChain) throws Exception {
+    Timer.Context timer = requestTimes.time();
+    requests.inc();
+    try {
+      return doAuthenticate(request, response, filterChain);
+    } catch(Exception e) {
+      numErrors.mark();
+      throw e;
+    } finally {
+      long elapsed = timer.stop();
+      totalTime.inc(elapsed);
+    }
+  }
 
   /**
    * Cleanup any per request  data
    */
   public void closeRequest() {
   }
 
+  @Override
+  public void initializeMetrics(SolrMetricManager manager, String registryName, String tag, final String scope) {
+    this.metricManager = manager;
+    this.registryName = registryName;
+    // Metrics
+    registry = manager.registry(registryName);
+    numErrors = manager.meter(this, registryName, "errors", getCategory().toString(), scope);
+    numTimeouts = manager.meter(this, registryName, "timeouts", getCategory().toString(), scope);
+    requests = manager.counter(this, registryName, "requests", getCategory().toString(), scope);
+    numAuthenticated = manager.counter(this, registryName, "authenticated", getCategory().toString(), scope);
+    numPassThrough = manager.counter(this, registryName, "passThrough", getCategory().toString(), scope);
+    numWrongCredentials = manager.counter(this, registryName, "failWrongCredentials", getCategory().toString(), scope);
```

Comment:
If auth does not succeed we log either wrong credentials, invalid credentials (e.g. malformed, if basicAuth header does not contain a ':'), missing credentials (if no header at all). Not all plugins will be able to distinguish between all these, such as hadoop plugin which I raised a question about in the JIRA. Should we have a more coarse-grained metric on this? Perhaps these are enough: `numAuthenticated`, `numPassThrough`, `numFailedAuth` and `numErrors`?

Reply:
I think that malformed credentials should be simply reported as errors.

Reply:
Done